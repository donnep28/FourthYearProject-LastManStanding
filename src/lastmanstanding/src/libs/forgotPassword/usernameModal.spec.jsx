import React from 'react';
import {shallow} from 'enzyme';
import UsernameModal from './usernameModal';

describe('usernameModal', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    getBack: jest.fn(),
    handleCodeSent: jest.fn(),
    username: 'mockUsername',
    setUsername: jest.fn(),
  };

  describe('Snapshot', () => {
    it('renders correctly', () => {
      const component = shallow(
          <UsernameModal />,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('interactions', () => {
    it('calls setUsername()', () => {
      const event = {target: {name: "username", value: "mockUsername"}};
      const component = shallow(
          <UsernameModal {...props}/>,
      );
      component.find("WithStyles(ForwardRef(TextField))").simulate('change', event);
      expect(props.setUsername).toHaveBeenCalled();
    });
    it('calls getBack()', () => {
        const component = shallow(
            <UsernameModal {...props}/>,
        );
        component.find("Link").simulate('click');
        expect(props.getBack).toHaveBeenCalled();
    });
  });
});