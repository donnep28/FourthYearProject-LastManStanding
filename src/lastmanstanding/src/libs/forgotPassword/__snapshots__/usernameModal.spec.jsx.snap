// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usernameModal Snapshot renders correctly 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-2"
    >
      <Memo(ForwardRef(LockOutlinedIcon)) />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Forgot Password
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      noValidate={true}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="uname"
        autoFocus={true}
        fullWidth={true}
        id="email"
        label="Username"
        margin="normal"
        name="Username"
        onChange={[Function]}
        required={true}
        variant="outlined"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-4"
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        type="submit"
        variant="contained"
      >
        Submit
      </WithStyles(ForwardRef(Button))>
    </form>
    <Link
      onClick={[Function]}
    >
      Back to Sign In
    </Link>
  </div>
</WithStyles(ForwardRef(Container))>
`;
